crates/polars-arrow/src/array/list/mod.rs

```diff
@@ -52,7 +52,7 @@ impl<O: Offset> ListArray<O> {
         let child_data_type = Self::try_get_child(&data_type)?.data_type();
         let values_data_type = values.data_type();
         if child_data_type != values_data_type {
-            polars_bail!(ComputeError: "ListArray's child's DataType must match. However, the expected DataType is {child_data_type:?} while it got {values_data_type:?}.");
+            polars_bail!(ComputeError: "ListArray's child's DataType must match, yet the expected DataType is {child_data_type:?} while it got {values_data_type:?}");
```

Comment:
This is too long for a single line error message. It should be broken up into a summary and a help text.